web/src/features/fba/pages/FireBehaviourAdvisoryPage.tsx

```diff
@@ -152,17 +154,55 @@ const FireBehaviourAdvisoryPage: React.FunctionComponent = () => {
     document.title = ASA_DOC_TITLE
   }, [])
 
+  const formControlRef = useRef<HTMLDivElement>(null)
+  const mapRef = useRef<HTMLDivElement>(null)
+  const navRef = useRef<HTMLDivElement>(null)
+  const sidePanelRef = useRef<HTMLDivElement>(null)
+  const [formControlHeight, setFormControlHeight] = useState<number>(0)
+  const [navRefHeight, setNavRefHeight] = useState<number>(0)
+
+  useEffect(() => {
+    if (navRef.current) {
+      const height = navRef.current.clientHeight
+      setNavRefHeight(height)
+    } // eslint-disable-line react-hooks/exhaustive-deps
+  }, [navRef.current?.clientHeight])
+
+  useEffect(() => {
+    if (formControlRef.current) {
+      const height = formControlRef.current.clientHeight
+      setFormControlHeight(height)
+    } // eslint-disable-line react-hooks/exhaustive-deps
+  }, [formControlRef.current?.clientHeight])
+
+  useEffect(() => {
```

Comment:
I think this `useEffect` can be merged with the one below it since they both don't specify a dependency array. Also is the intention for this to run once or on every render? If the former we should specify an empty dependency array: https://stackoverflow.com/a/57760937

Reply:
I can refactor it a bit to make it cleaner. My intention was to have that effect to run on every render, it was the only way I could get the map and panel to resize correctly when the form control changes sizes. 
I started with an empty dependency array which I quickly discovered didn't work with the way the page functions, I'm definitely open to there being a better way to do it though

Reply:
Got it! Makes sense thanks!